### PR TITLE
Support dependency-based workflow steps

### LINKED ISSUE #97: Support dependency-based step execution with after: [...]
## Goal
Allow workflows to declare dependency-based execution with `after: [...]` so steps can run once named prerequisite steps complete.

## Why
Squid Mesh is still mostly a sequential workflow runtime. Supporting step dependencies is the cleanest path to parallel fan-out and fan-in without pushing the project toward a visual builder.

## Acceptance Criteria
- `step/3` supports `after: [:step_a, :step_b]`
- validation rejects unknown dependencies and dependency cycles
- workflows can declare multiple entry steps when no dependencies are present
- the runtime only makes a step runnable after all declared dependencies complete successfully
- docs and at least one example workflow show the contract clearly

## Related
- Follow with #98 for runtime scheduling and join semantics
- Unblocks richer data flow work in #100
- Improves the inspection model needed in #101

### PR DESCRIPTION
**Pull request by cristianodabc**
## Summary

- Add dependency-based `after: [...]` workflow steps with compile-time validation, deterministic root ordering, and runtime progression support
- Fail closed when dependency resolution cannot produce a runnable next step, and add regression coverage for malformed graphs, ordering, and stuck-run edge cases
- Document the new workflow contract and cover it with the minimal host app example
- Closes #97

## Testing

- [x] `mix test`
- [x] `mix format --check-formatted`
- `mix test test/squid_mesh/workflow_test.exs test/squid_mesh/runtime/step_worker_test.exs`
- `cd examples/minimal_host_app && mix test test/workflow_runs_test.exs`
- `cd examples/minimal_host_app && MIX_ENV=test mix example.smoke`

## Checklist

- [x] Scope is focused and reviewable
- [x] Commits follow Conventional Commits
- [x] No secrets or machine-specific details were introduced

### CODE REVIEW

File: lib/squid_mesh/workflow/validation.ex [hunk 1]
```diff
@@ -35,24 +35,60 @@ defmodule SquidMesh.Workflow.Validation do
...
-      [entry_step] ->
-        entry_step
-
-      _other_steps ->
+      [] ->
         raise CompileError,
           file: env.file,
           line: env.line,
           description:
             "workflow validation failed:\n- workflow must define exactly one entry step"
```
[the remark this answers is not in the document]
Comment (cristianodabc) [reply lib/squid_mesh/workflow/validation.ex]: Addressed in `ba9ce34`. `entry_steps!/2` now raises a dependency-specific root-step error when dependency mode yields no entry roots.

File: lib/squid_mesh/workflow/validation.ex [hunk 2]
```diff
@@ -35,24 +35,60 @@ defmodule SquidMesh.Workflow.Validation do
...
+  """
+  @spec entry_step!(map(), Macro.Env.t()) :: atom() | nil
+  def entry_step!(definition, env) do
+    if dependency_mode?(definition.steps) do
+      nil
+    else
+      definition
+      |> entry_steps!(env)
+      |> List.first()
+    end
```
[the remark this answers is not in the document]
Comment (cristianodabc) [reply lib/squid_mesh/workflow/validation.ex:87]: Addressed in `ba9ce34`. The `entry_step!/2` doc now reflects the dependency-mode contract and explicitly documents the `nil` return.

File: lib/squid_mesh/runtime/step_executor/outcome.ex [hunk 3]
```diff
@@ -236,6 +263,41 @@ defmodule SquidMesh.Runtime.StepExecutor.Outcome do
...
+         context,
+         {:no_runnable_step, pending_steps}
+       ) do
+    case RunStore.transition_run(repo, run.id, :failed, %{
+           context: context,
+           current_step: run.current_step,
+           last_error: %{
+             message: "workflow step completed but no runnable next step was found",
+             failed_step: run.current_step,
+             pending_steps: pending_steps
```
[the remark this answers is not in the document]
Comment (cristianodabc) [reply lib/squid_mesh/runtime/step_executor/outcome.ex:278]: Addressed in `ba9ce34`. Public run-error deserialization now rewrites `pending_steps` through workflow step deserialization, so it is consistent with `failed_step` and `next_step`.
